File: Riley/src/types/calendar.ts
export interface CalendarEvent {
  id: string;
  title: string;
  description: string;
  startDate: string; // YYYY-MM-DD
  endDate: string; // YYYY-MM-DD
  allDay: boolean; // if true, no specific hours are highlighted
  startTime?: string; // HH:MM
  endTime?: string; // HH:MM
  location?: string;
  guests: string[]; // List of guest emails
  color: string; // Color key: e.g. 'purple', 'indigo', 'blue', 'pink', 'cyan', 'green', 'yellow', 'orange', 'red'
  permissions: {
    modify: boolean;
    inviteOthers: boolean;
    seeGuestList: boolean;
  };
}

export type ViewMode = 'month' | 'week' | 'day' | 'list';

export interface DateCell {
  date: Date;
  isCurrentMonth: boolean;
  isToday: boolean;
  events: CalendarEvent[];
}

File: RileyAI/src/types/calendar.py
from typing import List, Optional, Literal
from datetime import date, datetime, time
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel

class BaseCamelModel(BaseModel):
    """
    Base model config that automatically aliases snake_case fields 
    to camelCase to perfectly match the TypeScript frontend.
    """
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True
    )

class EventPermissions(BaseCamelModel):
    modify: bool
    invite_others: bool
    see_guest_list: bool

# ViewMode translates perfectly to a Python Literal
ViewMode = Literal['month', 'week', 'day', 'list']

class CalendarEvent(BaseCamelModel):
    id: str
    title: str
    description: str
    start_date: date  # Pydantic natively parses 'YYYY-MM-DD' strings
    end_date: date    
    all_day: bool
    start_time: Optional[time] = None  # Pydantic natively parses 'HH:MM' strings
    end_time: Optional[time] = None   
    location: Optional[str] = None
    guests: List[str]
    color: str  # e.g., 'purple', 'indigo', 'blue', etc.
    permissions: EventPermissions

class DateCell(BaseCamelModel):
    date: datetime  # JS 'Date' translates best to Python 'datetime'
    is_current_month: bool
    is_today: bool
    events: List[CalendarEvent]